service/app/queries/create_history_stream_test.go

```diff
@@ -13,61 +15,99 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestCreateHistoryStream_no_old_no_live(t *testing.T) {
-	a, err := di.BuildTestQueries()
-	require.NoError(t, err)
+const createHistoryStreamTestDelay = 1 * time.Second
+
+func TestCreateHistoryStream_IfOldAndLiveAreNotSetNothingIsWrittenAndStreamIsClosed(t *testing.T) {
+	t.Parallel()
 
 	ctx := fixtures.TestContext(t)
+	a := makeQueriesAndRunCreateHistoryStreamHandler(t, ctx)
+
+	rw := newCreateHistoryStreamResponseWriterMock()
 
 	query := queries.CreateHistoryStream{
-		Id:    fixtures.SomeRefFeed(),
-		Seq:   nil,
-		Limit: nil,
-		Live:  false,
-		Old:   false,
+		Id:             fixtures.SomeRefFeed(),
+		Seq:            nil,
+		Limit:          nil,
+		Live:           false,
+		Old:            false,
+		ResponseWriter: rw,
 	}
 
-	ch := a.Queries.CreateHistoryStream.Handle(ctx, query)
+	a.Queries.CreateHistoryStream.Handle(ctx, query)
 
-	for _ = range ch {
-		t.Fatal("channel should be closed right away")
-	}
+	<-time.After(createHistoryStreamTestDelay)
 
+	require.Equal(t, 1, a.MessagePubSub.CallsCount, "live should have been subscribed to in the background")
 	require.Empty(t, a.FeedRepository.GetMessagesCalls, "since old is not specified repository shouldn't have been called")
-	require.Zero(t, a.MessagePubSub.CallsCount, "since live is not specified pubsub shouldn't have been called")
+	require.Empty(t, rw.WrittenMessages)
+	require.Equal(t, []error{nil}, rw.WrittenErrors)
 }
 
-func TestCreateHistoryStream_if_repository_returns_error_live_messages_are_not_returned(t *testing.T) {
-	a, err := di.BuildTestQueries()
-	require.NoError(t, err)
+func TestCreateHistoryStream_IfOldIsSetNothingIsWrittenAndStreamIsClosed(t *testing.T) {
+	t.Parallel()
+
+	ctx := fixtures.TestContext(t)
+	a := makeQueriesAndRunCreateHistoryStreamHandler(t, ctx)
+
+	rw := newCreateHistoryStreamResponseWriterMock()
+
+	query := queries.CreateHistoryStream{
+		Id:             fixtures.SomeRefFeed(),
+		Seq:            nil,
+		Limit:          nil,
+		Live:           false,
+		Old:            true,
+		ResponseWriter: rw,
+	}
+
+	a.Queries.CreateHistoryStream.Handle(ctx, query)
+
+	<-time.After(createHistoryStreamTestDelay)
+
+	require.Equal(t, 1, a.MessagePubSub.CallsCount, "live should have been subscribed to in the background")
+	require.NotEmpty(t, a.FeedRepository.GetMessagesCalls)
+	require.Empty(t, rw.WrittenMessages)
+	require.Equal(t, []error{nil}, rw.WrittenErrors)
```

Comment:
I'm worried that waiting for a fixed duration can lead to flaky or at least unreliable tests in the future as we cannot be sure that everything will be executed during that time. 

Do you think we could make it `require.Eventually` and wait for all conditions to be satisfied? Probably we could even improve `ResponseWriterMock` API so it can tell us that `CloseWithErr` was called or not. Having that we can wait for closing in `require.Eventually`, and after that verify all remaining conditions.